### PR TITLE
Fix the error message of tsdata call sequence

### PR DESCRIPTION
**Pull request by liangs6212**

### CODE REVIEW

**Comment by liangs6212** [comment]
#4100

File: pyzoo/zoo/chronos/data/tsdataset.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
 from zoo.chronos.data.utils.resample import resample_timeseries_dataframe
 from zoo.chronos.data.utils.split import split_timeseries_dataframe
 
+from sklearn.utils.validation import check_is_fitted
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/data/tsdataset.py:29]: I think you can move this import to line 520 (right before your calling). This **may** prevent some version issue in the future.

File: pyzoo/zoo/chronos/data/tsdataset.py [hunk 2]
```diff
@@ -262,6 +262,8 @@ def gen_global_feature(self, settings="comprehensive", full_settings=None):
         :return: the tsdataset instance.
 
         '''
+        assert not self._feature_generation_completed, "We think get_feature and \
+            gen_rolling do not need to be called multiple times, please remove redundant calls."
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/data/tsdataset.py:266]: "Only one of gen_global_feature and gen_rolling_feature should be called."

File: pyzoo/zoo/chronos/data/tsdataset.py [hunk 3]
```diff
@@ -56,11 +56,11 @@ def __init__(self, data, **schema):
         self.roll_feature = None
         self.roll_target = None
         self.roll_feature_df = None
-        self.roll_addional_feature = None
+        self.roll_addtional_feature = None
         self.scaler = None
         self.scaler_index = [i for i in range(len(self.target_col))]
         self.id_sensitive = None
-
+        self._feature_generation_completed = False
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/data/tsdataset.py:63]: self._has_generate_agg_feature

File: pyzoo/zoo/chronos/data/tsdataset.py [hunk 4]
```diff
@@ -310,6 +312,8 @@ def gen_rolling_feature(self,
 
         :return: the tsdataset instance.
         '''
+        assert not self._feature_generation_completed, "We think get_feature and \
+            gen_rolling do not need to be called multiple times, please remove redundant calls."
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/data/tsdataset.py:316]: "Only one of gen_global_feature and gen_rolling_feature should be called."

File: pyzoo/zoo/chronos/data/tsdataset.py [hunk 5]
```diff
@@ -56,11 +56,11 @@ def __init__(self, data, **schema):
         self.roll_feature = None
         self.roll_target = None
         self.roll_feature_df = None
-        self.roll_addional_feature = None
+        self.roll_addtional_feature = None
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/data/tsdataset.py:59]: additional

**Review by TheaperDeng — APPROVED** [review]
LGTM, pls run jenkins

**Comment by liangs6212** [comment]
http://10.239.176.111:18888/job/ZOO-PR-Python-ExampleTests-py37-ray-horovod-Spark3/1500/
http://10.239.176.111:18888/job/ZOO-PR-Validation/6079/

File: pyzoo/test/zoo/chronos/data/test_tsdataset.py [hunk 6]
```diff
@@ -532,3 +531,21 @@ def test_tsdataset_rolling_feature_multiple(self):
...
+                                                            id_col="id",
+                                                            with_split=True,
+                                                            val_ratio=0.1,
+                                                            test_ratio=0.1)
+        from sklearn.preprocessing import StandardScaler
+        stand = StandardScaler()
+        with pytest.raises(AssertionError):
+            for tsdata in [td_train, td_valid, td_test]:
+                tsdata.scale(stand, fit=False)
+            tsdata._check_basic_invariants()
```
Comment (TheaperDeng) [inline pyzoo/test/zoo/chronos/data/test_tsdataset.py:551]: add a test for call both `gen_global` and `gen_rolling`.

File: pyzoo/zoo/chronos/data/tsdataset.py [hunk 7]
```diff
@@ -508,15 +512,17 @@ def scale(self, scaler, fit=True):
...
             for feature in self.feature_col:
-                if feature not in self.roll_addional_feature:
+                if feature not in self.roll_additional_feature:
                     feature_col.append(feature)
         if fit:
             self.df[self.target_col + feature_col] = \
                 scaler.fit_transform(self.df[self.target_col + feature_col])
         else:
+            assert [v for v in vars(scaler) if v.endswith("_") and not v.startswith("__")],\
+                "When calling scale for the first time, you need to set fit=True."
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/data/tsdataset.py:525]: I think the prev one is better? This one is kind of "hacking"

**Review by TheaperDeng — CHANGES_REQUESTED** [review]

File: pyzoo/zoo/chronos/data/tsdataset.py (hunk 7 above)
Comment (liangs6212) [reply pyzoo/zoo/chronos/data/tsdataset.py]: Has become our shape.

**Review by TheaperDeng — APPROVED** [review]
LGTM, pls run jenkins

**Comment by liangs6212** [comment]
http://10.112.231.51:18888/job/ZOO-PR-Python-Spark-3.0-py37-ray/1641/
http://10.112.231.51:18888/job/ZOO-PR-Python-ExampleTests/3475/
http://10.112.231.51:18888/job/ZOO-PR-Scala-StyleCheck/6270/
http://10.112.231.51:18888/job/ZOO-PR-Python-Spark-2.4-py37-ray/2439/
http://10.112.231.51:18888/job/ZOO-PR-Python-Spark-2.3-py37/3005/
http://10.112.231.51:18888/job/ZOO-PR-Validation/6169/

**Comment by liangs6212** [comment]
http://10.112.231.51:18888/job/ZOO-PR-Validation/6193/
http://10.112.231.51:18888/job/ZOO-PR-Python-Spark-3.1-py36k216/1635/
http://10.112.231.51:18888/job/ZOO-PR-Python-Spark-3.1-py36-ray/1642/

**Pull request merged**